CMakeLists.txt

```diff
@@ -107,10 +108,20 @@ if (NOT CMAKE_BUILD_TYPE)
   message ("No build type detected. Build type set to Release by default")
   set ( CMAKE_BUILD_TYPE Release )
 endif()
-
-set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-rtti -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DVC_DEBUG -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -fPIC -fvisibility-inlines-hidden" )
+set ( CMAKE_CXX_STANDARD 14 )
+set ( CMAKE_CXX_STANDARD_REQUIRED ON )
+set ( CMAKE_CXX_EXTENSIONS OFF )
+set ( CMAKE_VISIBILITY_INLINES_HIDDEN ON )
+set ( CMAKE_POSITION_INDEPENDENT_CODE ON )
+# https://gcc.gnu.org/wiki/Visibility#line-14 - Much lower chance of symbol collision.
+set ( CMAKE_CXX_VISIBILITY_INLINES_HIDDEN ON )
+# set ( CMAKE_INTERPROCEDURAL_OPTIMIZATION ON) 
+# CMAKE_INTERPROCEDURAL_OPTIMIZATION applies different optimizations depending on the compiler. It breaks cross compiler compatibility (e.g. g++ with clang)
```

Comment:
That requires a bit further investigation/explanation. Which target breaks compatibility with which other?

Reply:
https://github.com/tdlib/td/issues/1244#issuecomment-727657360
LTO is platform and compiler dependent, if the library is being used with another compiler breaks most stuff (gnu compilers and clang clash with weird errors such as :
```
/usr/bin/ld: /usr/local/lib/libVersioningCompiler.a: error adding symbols: file format not recognized
collect2: error: ld returned 1 exit status```